pkg/cmd/cli/cmd/idle.go

```diff
@@ -0,0 +1,630 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	utilerrors "github.com/openshift/origin/pkg/util/errors"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployclient "github.com/openshift/origin/pkg/deploy/client/clientset_generated/internalclientset/typed/core/unversioned"
+	unidlingapi "github.com/openshift/origin/pkg/unidling/api"
+	utilunidling "github.com/openshift/origin/pkg/unidling/util"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	clientset "k8s.io/kubernetes/pkg/client/unversioned/adapters/internalclientset"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/strategicpatch"
+)
+
+const (
+	idleLong = `
+Idle scalable resources.
+
+Idling discovers the scalable resources (such as deployment configs and replication controllers)
+associated with a series of services by examining the endpoints of the service.
+Each service is then marked as idled, the associated resources are recorded, and the resources
+are scaled down to zero replicas.
+
+Upon receiving network traffic, the services (and any associated routes) will "wake up" the
+associated resources by scaling them back up to their previous scale.`
+
+	idleExample = `  # Idle the scalable controllers associated with the services listed in to-idle.txt
+  $ %[1]s idle -f to-idle.txt`
+)
+
+// NewCmdStatus implements the OpenShift cli status command
+func NewCmdIdle(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	o := &IdleOptions{
+		out:    out,
+		errOut: errOut,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "idle -f FILENAME",
+		Short:   "Idle scalable resources",
+		Long:    idleLong,
+		Example: fmt.Sprintf(idleExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			err := o.RunIdle(f)
+			if err == cmdutil.ErrExit {
+				os.Exit(1)
+			}
+			kcmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().BoolVar(&o.dryRun, "dry-run", false, "If true, only print the annotations that would be written, without annotating or idling the relevant objects")
+	cmd.Flags().StringVarP(&o.filename, "filename", "f", o.filename, "file containing list of services whose scalable resources to idle")
```

Comment:
I missed this before, but we shouldn't be using `filename` for this since it's not the same as all the other uses of `-f`.  We should call this something different, like `resource-names-file` (it doesn't need a short cut flag).